packages/e2e-tests/specs/block-deletion.test.js

```diff
@@ -108,4 +109,29 @@ describe( 'block deletion -', () => {
 			expect( await getEditedPostContent() ).toMatchSnapshot();
 		} );
 	} );
+
+	describe( 'deleting all paragraphs', () => {
+		it( 'results in the default block getting selected', async () => {
+			// Add a third paragraph for this test.
+			await page.keyboard.type( 'Third paragraph' );
+
+			// Move the mouse to show the block toolbar
+			await page.mouse.move( 200, 300, { steps: 10 } );
+
+			await clickOnBlockSettingsMenuItem( 'Remove Block' );
```

Comment:
I feel like the `beforeEach` common behavior in this suite is making this test more difficult than it needs to be. It seems like it should just be almost as simple as:

```js
await insertBlock( 'Paragraph' );
await page.keyboard.press( 'Backspace' );
```

...followed by the concluding assertions.

Reply:
> It seems like it should just be almost as simple as:

In fact, as part of the reimplementation I considered at https://github.com/WordPress/gutenberg/pull/14189#issuecomment-468779914 , the "Backspace" behavior is distinct, since it's actually relying on the `REPLACE_BLOCKS` effect handler to ensure the default block. A test for it should probably also exist, but it's technically a different behavior than what we're fixing with #14189.